From aad3e3d4d88d36b3a5d886515681ca18840d993e Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Tue, 21 Jun 2016 21:19:37 +0800
Subject: [PATCH 1226/1240] fix: pm: a3700: update target CPU frequency divider

The target CPU frequency divider values are modified
when the original CPU divider is 2.
The divider values for four CPU frequency loads are
set to 2, 4, 5, 6 accordingly.

Change-Id: Ib764839d3c13ef1076b5f1db618edd545977a973
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30597
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
---
 arch/arm/cpu/armv8/armada3700/pm.c | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/arch/arm/cpu/armv8/armada3700/pm.c b/arch/arm/cpu/armv8/armada3700/pm.c
index 9a7ebe3..9cfa48d 100644
--- a/arch/arm/cpu/armv8/armada3700/pm.c
+++ b/arch/arm/cpu/armv8/armada3700/pm.c
@@ -74,15 +74,27 @@ enum dvfs_load_index {
 #define PM_MAX_DT_NUM	1
 struct pm_config __attribute__((section(".data"))) g_pm_config;
 
-/* CPU clock TBG divider array
+/*
+ * CPU clock TBG divider array
  * TBG divider 0 and 7 mean active-high for clock output,
- * thus default divider 2 are kept for them for all the four levels.
+ * thus they are not listed below.
  * For other divider values from 1 to 6, proper divider values should be set
  * for each level from small to big.
+ * Recommended divider values for real deployment:
+ * original divider   1st divider    2nd divider   3rd divider   4th divider
+ *   1                  1              2             4             6
+ *   2                  2              4             5             6
+ * CPU frequency mapping:
+ * original CPU freq    1st CPU freq     2nd CPU freq    3rd CPU freq    4th CPU freq
+ *   1200M                1200M             600M             300M            200M
+ *   1000M                1000M             500M             250M            166M
+ *   800M                 800M              400M             200M            133M
+ *   600M                 600M              300M             240M            200M
+ *   400M                 400M              200M             160M            133M
  */
 enum a3700_tbg_divider tbg_div_arr[TBG_DIVIDER_NUM][DVFS_LOAD_MAX_NUM] = {
 	{TBG_DIVIDER_1, TBG_DIVIDER_2, TBG_DIVIDER_4, TBG_DIVIDER_6},
-	{TBG_DIVIDER_2, TBG_DIVIDER_3, TBG_DIVIDER_4, TBG_DIVIDER_6},
+	{TBG_DIVIDER_2, TBG_DIVIDER_4, TBG_DIVIDER_5, TBG_DIVIDER_6},
 	{TBG_DIVIDER_3, TBG_DIVIDER_4, TBG_DIVIDER_5, TBG_DIVIDER_6},
 	{TBG_DIVIDER_4, TBG_DIVIDER_4, TBG_DIVIDER_5, TBG_DIVIDER_6},
 	{TBG_DIVIDER_5, TBG_DIVIDER_5, TBG_DIVIDER_6, TBG_DIVIDER_6},
-- 
1.9.1

